stdlib/public/Concurrency/Executor.swift

```diff
@@ -81,12 +101,26 @@ public struct UnownedSerialExecutor: Sendable {
 @_silgen_name("swift_task_isOnExecutor")
 public func _taskIsOnExecutor<Executor: SerialExecutor>(_ executor: Executor) -> Bool
 
+/// Primarily a debug utility.
+///
+/// If the passed in Job is a Task, returns the complete 64bit TaskId,
+/// otherwise returns only the job's 32bit Id.
+///
+/// - Returns: the Id stored in this Job or Task, for purposes of debug printing
+@available(SwiftStdlib 5.9, *)
+@_silgen_name("swift_task_getJobTaskId")
+internal func _getJobTaskId(_ job: UnownedJob) -> UInt64
+
 // Used by the concurrency runtime
 @available(SwiftStdlib 5.1, *)
 @_silgen_name("_swift_task_enqueueOnExecutor")
-internal func _enqueueOnExecutor<E>(job: UnownedJob, executor: E)
+internal func _enqueueOnExecutor<E>(job unownedJob: UnownedJob, executor: E)
 where E: SerialExecutor {
-  executor.enqueue(job)
+  if #available(SwiftStdlib 5.9, *) {
+    executor.enqueueJob(Job(context: unownedJob.context))
+  } else {
+    executor.enqueue(unownedJob)
+  }
```

Comment:
Attempting this dance to call the new method when we can directly -- the new method has a default impl that directs to the old one. So old impls continue to work ok.